src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -1217,6 +1231,23 @@ public override SyntaxNode InsertNamespaceImports(SyntaxNode declaration, int in
             }
         }
 
+        private static SyntaxList<UsingDirectiveSyntax> CopyBanner(CompilationUnitSyntax cu, SyntaxList<UsingDirectiveSyntax> usingsToInsert)
+        {
+            // First. Strip any pp directives or banners on the compilation unit.  They
+            // have to stay at the top of the list.
+            IEnumerable<SyntaxTrivia> banner = cu.GetLeadingBannerAndPreprocessorDirectives();
+
+            // Now, we want to remove any blank lines from the new first node and then reattach the banner. 
+            var finalFirstNode = usingsToInsert[0];
+            finalFirstNode = finalFirstNode.GetNodeWithoutLeadingBlankLines();
```

Comment:
Why don't you use the PreservieTrivia pattern.  Maybe a PerserveBanner API with a callback with the CU with the banner text stripped, that gets reattached when the callback returns?